### PR TITLE
Add extension to verify role bindings for user 

### PR DESCRIPTION
**Pull request by joesims22**
This PR adds the extension VerifyRoleBindingsForUser which validates the corresponding role bindings have been created for a user.

### CODE REVIEW

File: extensions/rbac/verify.go [hunk 1]
```diff
@@ -46,6 +49,27 @@ func VerifyGlobalRoleBindingsForUser(t *testing.T, user *management.User, adminC
 	assert.Equal(t, 1, len(grbs.Data))
 }
 
+// VerifyRoleBindingsForUser validates that the corresponding role bindings are created for the user
+func VerifyRoleBindingsForUser(t *testing.T, user *management.User, standardClient *rancher.Client, clusterID, role string) {
+	rblist, err := rbac.ListRoleBindings(standardClient, localcluster, clusterID, metav1.ListOptions{})
```
Comment (anupama2501) [inline extensions/rbac/verify.go:54]: shouldn't the client be admin client? standard user client won't be able to list role bindings from the local cluster.

File: extensions/rbac/verify.go [hunk 2]
```diff
@@ -46,6 +49,27 @@ func VerifyGlobalRoleBindingsForUser(t *testing.T, user *management.User, adminC
...
+	require.NoError(t, err)
+	userID := user.Resource.ID
+	userRoleBindings := []string{}
+
+	for _, rb := range rblist.Items {
+		if rb.Subjects[0].Kind == "User" && rb.Subjects[0].Name == userID {
```
Comment (anupama2501) [inline extensions/rbac/verify.go:60]: Can we add "User" as a constant?

**Review by anupama2501 — COMMENTED** [review]
looks good, have a few comments. Do we have a jenkins log for this ?

File: extensions/rbac/verify.go (hunk 1 above)
Comment (joesims22) [reply extensions/rbac/verify.go:54]: Yes that's correct, I'll change it to admin.

File: extensions/rbac/verify.go (hunk 2 above)
Comment (joesims22) [reply extensions/rbac/verify.go:60]: Will add User as a constant 👍

File: extensions/rbac/verify.go [hunk 3]
```diff
@@ -34,6 +36,8 @@ const (
 	standardUser                  = "user"
 	activeStatus                  = "active"
 	forbiddenError                = "403 Forbidden"
+	localcluster                  = "local"
```
Comment (Priyashetty17) [inline extensions/rbac/verify.go:39]: nit., Can you change the variable name to `localCluster`?

**Review by Priyashetty17 — COMMENTED** [review]
lgtm, just a nit on the variable name

File: extensions/rbac/verify.go (hunk 3 above)
Comment (joesims22) [reply extensions/rbac/verify.go]: changed constant name 👍

**Review by anupama2501 — APPROVED** [review]
lgtm

File: extensions/rbac/verify.go [hunk 4]
```diff
@@ -46,6 +50,27 @@ func VerifyGlobalRoleBindingsForUser(t *testing.T, user *management.User, adminC
 	assert.Equal(t, 1, len(grbs.Data))
 }
 
+// VerifyRoleBindingsForUser validates that the corresponding role bindings are created for the user
+func VerifyRoleBindingsForUser(t *testing.T, user *management.User, adminClient *rancher.Client, clusterID, role string) {
+	rblist, err := rbac.ListRoleBindings(adminClient, localcluster, clusterID, metav1.ListOptions{})
```
Comment (Priyashetty17) [inline extensions/rbac/verify.go:55]: Can you update the constant here?

Comment (joesims22) [reply extensions/rbac/verify.go]: updated 👍

**Review by Priyashetty17 — APPROVED** [review]
lgtm

**Review by igomez06 — COMMENTED** [review]
My question for this and maybe @anupama2501 can help answer it. Why do we need this to be a verification in the framework. Can't it just go in the test itself?

**Comment by igomez06** [comment]
> @igomez06 this was part of the rbac validations we moved from validation suite to extension suite as part of this task [rancher/qa-tasks#892](https://github.com/rancher/qa-tasks/issues/892)

This should only be the case if the verification is being used outside of the package, if not we're unnecessarily bloating the framework when this func should just be in the test. So you're saying VerifyRoleBindingsForUser is being used outside of tests/v2/validation/rbac?

**Comment by anupama2501** [comment]
sorry @igomez06 for the delayed response. Yes, this is to make use of the role bindings verification outside of `VerifyRoleBindingsForUser` as well. We have a few tests in rbac additional for which we can make use of this function.

However, considering this involves actions rather than verification, it is appropriate for actions. Since we already have a PR for this, we can merge it now and later move it to actions once that is established. Let me know if that sounds good.

**Comment by igomez06** [comment]
> sorry @igomez06 for the delayed response. Yes, this is to make use of the role bindings verification outside of `VerifyRoleBindingsForUser` as well. We have a few tests in rbac additional for which we can make use of this function.
> 
> However, considering this involves actions rather than verification, it is appropriate for actions. Since we already have a PR for this, we can merge it now and later move it to actions once that is established. Let me know if that sounds good.

Yes this would be moved any way, but my question remains. Is this function going to be used outside of tests/v2/validation/rbac? If not, then I would argue that this should just go into tests/v2/validation/rbac/rbac.go

**Comment by anupama2501** [comment]
Yes, once the RBAC tests are relocated to the integration suite, they will be referenced outside of the tests/v2/validation/ directory. In that scenario, this should be maintained in actions.

**Review by igomez06 — APPROVED** [review]
LGTM

**Comment by anupama2501** [comment]
Yes @igomez06 will wait on actions to be implemented and once thats done this can be ported over to actions. We can then close this PR.

**Comment by joesims22** [comment]
This code has been ported over to actions [#47447](https://github.com/rancher/rancher/pull/47447)

Closing this since code will reside in rancher/rancher actions.

**Pull request closed without merging**